persistence/src/test/java/cz/fi/muni/pa165/dao/VisitDaoTest.java

```diff
@@ -0,0 +1,282 @@
+package cz.fi.muni.pa165.dao;
+
+import cz.fi.muni.pa165.PersistenceApplicationContext;
+import cz.fi.muni.pa165.entity.*;
+import cz.fi.muni.pa165.enums.MushroomType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.testng.Assert;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceUnit;
+import javax.transaction.Transactional;
+import java.util.Date;
+import java.util.List;
+/**
+ * @author Jiri Sacha 409861
+ */
+
+@ContextConfiguration(classes = PersistenceApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@org.springframework.transaction.annotation.Transactional
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
```

Comment:
this is already done with this http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/test/context/transaction/TransactionalTestExecutionListener.html + @Transactional